docs/how-to-use-with-rails-backend.md

```diff
@@ -0,0 +1,169 @@
+
+By default, spraygun initializes a git repository in the generated directory. This will conflict with the git repository that is already in place for the Rails app. It must be removed before proceeding.
+
+```
+$ rm -rf client/.git
+```
+
+### 4. Change the default port of the React app
+
+In development, the React app runs on port 3000 by default. This conflicts with Rails, which also runs on that port. To avoid this conflict, configure the React app to use port 5000 instead.
+
+Create a `client/.env.development` file with these contents:
+
+```
+PORT=5000
+```
+
+Verify that the React app boots on port 5000.
+
+```
+$ cd client
+$ yarn start
+```
+
+Your browser should open to `http://localhost:5000`.
+
+### 5. Set up the development proxy
+
+The React development server (i.e. the one running on port 5000) can automatically proxy API requests from the React frontend to the Rails backend running on port 3000. This makes for a seamless development experience with no CORS configuration necessary.
+
+Edit `client/package.json` and add the following top-level entry:
+
+```
+"proxy": "http://localhost:3000"
+```
+
+### 6. Add a root package.json
+
+Adding a `package.json` in the root of the Rails project, provides a couple of nice conveniences:
+
+- install `client` dependencies when a developer (or Heroku) runs `yarn install` at the top level
+- start the Rails server and React client in a single `yarn start` command
+
+Create a `package.json` at the root of the Rails project with these contents:
+
+```json
+{
+  "name": "my-project",
+  "version": "1.0.0",
+  "private": true,
+  "scripts": {
+    "start": "concurrently -k -r npm:start:client npm:start:server",
+    "start:client": "cd client && npm-run-all start",
+    "start:server": "heroku local",
+    "postinstall": "cd client && yarn install"
+  },
+  "devDependencies": {
+    "concurrently": "^3.6.1",
+    "heroku": "^7.7.7",
+    "npm-run-all": "^4.1.3"
+  },
+  "engines": {
+    "node": "8.11.4",
+    "yarn": "1.9.4"
+  },
+  "cacheDirectories": ["node_modules", "client/node_modules"]
+}
+```
+
+Then run this at the top level:
+
+```
+$ yarn install
+```
+
+You can now start both the server and the client with one command:
+
+```
+$ yarn start
+```
+
+### 7. Add a rake task to build React
+
+When Rails is deployed (on Heroku or any other platform), the `assets:precompile` task is invoked. To ensure that the React frontend is deployed alongside the Rails backend, hook into the precompile step to build the React app and copy it into the appropriate place where Rails can serve it.
+
+Create a `lib/tasks/react.rake` file with these contents:
+
+```ruby
+namespace :react do
+  desc "Builds the React app from client/ into the Rails public/ directory"
+  task :build do
+    FileUtils.cd("client") do
+      system("yarn build") || raise("Failed to build React app")
+      FileUtils.cp_r("build/.", "../public", verbose: true)
+    end
+  end
+end
+
+task "assets:precompile" do
```

Comment:
I wonder if this clashes with the existing asset compilation task. It's not an unrealistic scenario to use Sprockets and Webpack in the same app.

Maybe this ends up as:

```ruby
Rake::Task["assets:precompile"].enhance do
  Rake::Task["react:build"].invoke
end
```

Reply:
Rake tasks are additive (it is one of the many confusing parts of Rake), so defining a task with the same name actually enhances the existing task instead of replacing it.

I like your idea of using `enhance` because it makes this side-effect explicit. 👍